waku/v2/discovery_connector.go

```diff
@@ -245,8 +245,18 @@ func (c *PeerConnectionStrategy) workPublisher(ctx context.Context) {
 	}
 }
 
+const maxActiveDials = 5
+
 func (c *PeerConnectionStrategy) dialPeers(ctx context.Context) {
 	defer c.wg.Done()
+
+	maxGoRoutines := c.minPeers
+	if maxGoRoutines > maxActiveDials {
+		maxGoRoutines = maxActiveDials
+	}
+
+	sem := make(chan struct{}, maxGoRoutines)
```

Comment:
Why not use a channel of int?

Reply:
This is a buffered signal channel used as a semaphore to limit the number of go-routines. I used struct{} because an empty struct has no memory cost.

I could have used "golang.org/x/sync/semaphore" instead but using this is IMO simpler and achieves the same result

Reply:
This is the better option then.